core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java

```diff
@@ -97,6 +97,7 @@
 
   @Inject SharedPreferenceUtil sharedPreferenceUtil;
   @Inject ZimReaderContainer zimReaderContainer;
+  @Inject protected AlertDialogShower alertDialogShower;
```

Comment:
If @Inject is added, the app does not build.

Reply:
For a very good reason, DialogModule should not be added to CoreComponent

Look at AlertDialogShower
`class AlertDialogShower @Inject constructor(private val activity: Activity) ` 
it needs an Activity to be constructed.

DialogModule is unfortunately a misdirection, this class should be deleted, it is not used anywhere.

ActivityModule is what you should be looking at because that is where we provide the binding of AlertDialogShower to the interface `DialogShower`

```
  @Binds
  @ActivityScope
  abstract fun bindDialogShower(alertDialogShower: AlertDialogShower): DialogShower
```

Because this class resides in core module we should look to `CoreActivityComponent`

When you create a `CoreActivityComponent` you pass in an activity
```
  @Subcomponent.Builder
  interface Builder {
    @BindsInstance
    fun activity(activity: Activity): CoreActivityComponent.Builder

    fun build(): CoreActivityComponent
  }
```

so whatever we define in the modules, if they need an activity they will get this one.

Now the issue becomes how do you create this component. I'll leave it as an exercise

Reply:
Finally, I guess I have cracked it 😃 . Please check  it once.